pkg/api/doctor.go

```diff
@@ -0,0 +1,118 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"golang-training-Hospital/pkg/data"
+	"log"
+	"net/http"
+	"strconv"
+)
+
+//DoctorApi - struct that's represent api layer between data layer and user layer
+type DoctorApi struct {
+	data *data.DoctorData
+}
+
+//ServeDoctorResources - create a system of routing in web application
+func ServeDoctorResources(r *mux.Router, data data.DoctorData) {
+	api := &DoctorApi{data: &data}
+	r.HandleFunc("/doctors", api.getAllDoctors).Methods("GET")
+	r.HandleFunc("/create", api.createDoctor).Methods("POST")
+	r.HandleFunc("/update", api.updateDoctor).Methods("PUT")
+	r.HandleFunc("/delete", api.deleteDoctor).Methods("DELETE")
+}
+
+//getAllDoctors - get all records from doctors table
+//writer - perform response object from the server.The type is http.ResponseWriter
+//request - perform request from the user.The type is *http.Request
+func (a DoctorApi) getAllDoctors(writer http.ResponseWriter, request *http.Request) {
```

Comment:
why we should use private methods here? Let's make `getAllDoctors`, `createDoctor`, `updateDoctor`, `deleteDoctor` - public